common/utils/budget.js

```diff
@@ -661,6 +662,38 @@ export const getCostFromItemByYear = (item, year) => {
 export const getPropCostType = category =>
   category === 'contractors' ? 'contractors' : 'inHouse';
 
+/**
+ * Add cost share totals of each category (medicaid, federal, state) to
+ * the budget's combined total for the particular year and the total of all the years.
+ * @param {Object} budget The budget object
+ * @param {Number} year The FFY
+ * @param {Number} medicaidShare The medicaid share are total costs to be paid by both federal and state
+ * @param {Number} fedShare The federal share of the costs to be paid
+ * @param {Number} stateShare The state share of the costs to be paid
+ * @returns {Object} the updated budget
+ */
+export const sumShareCostsForCombinedTotals = ({
+  budget,
+  year,
+  medicaidShare = 0,
+  fedShare = 0,
+  stateShare = 0
+} = {}) => {
+  const updatedBudget = deepCopy(budget);
+  if (year) {
+    // add to combined (total for all funding) for the particular year
+    updatedBudget['combined'][year].medicaid += medicaidShare;
```

Comment:
we need to be storing a value for total here too

Reply:
Yes, definitely. All the "total" values are actually getting added by an existing function `sumTotalCostsByCategory`, so we don't need to do it in this function. But when I refactor to include the split, I'll make sure that's still the case and if not, I'll add the totals explicitly here.